### ISSUE TITLE
Weapon Specialist List Doesn't Differentiate Between Inner Sphere and Clan Weapons 0.49.8

### ISSUE THREAD

**Issue by BLR-IIC** [MegaMek/megamek#3876]
This has been a problem for as long as I can remember the Weapon Specialist SPA being a thing in MegaMek.  It's not obvious which same-named weapons on the list are Inner Sphere or Clan.  This is true with all ER energy weapons, standard pulse lasers, ultra ACs, rotary ACs, LBX ACs, LRMs, and SRMs.  Oddly enough all LRT and SRT weapons are missing from the list.  A player doesn't find out whether their Weapon Specialist SPA choice is valid until the game begins.  Then it's too late to make any changes to fix it.  This is especially problematic when MekHQ automatically assigns an invalid Weapon Specialist SPA for the setting, like a specialist of Clan SRM-2s for an Inner Sphere Succession Wars merc company.

**Comment by Thom293** [comment]
Just curious - is the SPA you are talking about assigned on Company Creation in MHQ, or on an Open Market pilot, or in MM itself?

I looked at some of this code, but I think it exists in a couple of different places.  Just trying to understand where the one you are talking about arises.  The one I looked at checks the weapons on the mech, I think, before it assigns an SPA.  But that may not be the one you are talking about.

**Comment by pheonixstorm** [comment]
If you edit a pilot in HQ and assign the WS SPA it gives you a dropdown menu that you have to manually select the weapon you want to assign. It may be the same in MM as well, I haven't tried.

During normal assignment in HQ (by spending XP) it only allows you to choose from the weapons on your assigned unit.

**Comment by BLR-IIC** [comment]
I just tried this in just MegaMek 0.49.8 and the weapon specialist SPA in the Configure Unit menu only shows the weapons equipped by the unit.  In MekHQ's GM > Edit person menu, the Weapon Specialist SPA shows _all_ weapons in the manner I described above.  Perhaps this is actually more of a MekHQ issue then?  For some reason I thought it was also a MegaMek issue, but that seems to have been addressed a long time ago though I don't see a ticket for it on GitHub.

**Comment by Thom293** [comment]
Yeah I was just wondering if you were using the unit generator in MHQ or the Force Generator In MM or it was just a random pilot off the market with a unit or created by the OpFor generator in stratcon.  It was told to me at one point that this is some rough area of code.  Im not sure if it is 3 separate modules, or if there is any cross-cooperation among them.
